apps/src/applab/designMode.js

```diff
@@ -638,6 +638,62 @@ designMode.onDuplicate = function(element, prevThemeName, event) {
 
 var batchChangeId = 1;
 
+designMode.onRestoreThemeDefaults = function(element) {
+  firehoseClient.putRecord({
+    study: FIREHOSE_STUDY,
+    study_group: FIREHOSE_GROUP,
+    event: 'restore_theme_defaults',
+    project_id: project.getCurrentId(),
+    data_json: JSON.stringify({
+      elementId: element.id,
+      elementTag: element.tagName,
+      elementClass: element.className
+    })
+  });
+
+  const currentThemeValue = elementLibrary.getCurrentTheme(
+    designMode.activeScreen()
+  );
+  const themeValues = elementLibrary.getThemeValues(element);
+  let modifiedProperty = false;
+  // Start a new batched set of updateProperty() calls:
+  batchChangeId++;
+  for (const propName in themeValues) {
+    const dataModifiedAttributeName = `data-mod-${propName}`;
```

Comment:
How does dataModifiedAttributeName get used? I would have expected to read this attribute here, but instead, we just remove it.

Reply:
It gets set when themes are changed: see `changeThemeForElement()`: https://github.com/code-dot-org/code-dot-org/blob/staging/apps/src/applab/designMode.js#L641